watertap/unit_models/electroNP_ZO.py

```diff
@@ -0,0 +1,494 @@
+            doc="Solute removal fraction on a mass basis",
+        )
+
+        # Add performance constraints
+        # Water recovery
+        @self.Constraint(self.flowsheet().time, doc="Water recovery equation")
+        def water_recovery_equation(b, t):
+            return b.recovery_frac_mass_H2O[t] * b.properties_in[
+                t
+            ].get_material_flow_terms("Liq", "H2O") == b.properties_treated[
+                t
+            ].get_material_flow_terms(
+                "Liq", "H2O"
+            )
+
+        # Flow balance
+        @self.Constraint(self.flowsheet().time, doc="Overall flow balance")
+        def water_balance(b, t):
+            return b.properties_in[t].get_material_flow_terms(
+                "Liq", "H2O"
+            ) == b.properties_treated[t].get_material_flow_terms(
+                "Liq", "H2O"
+            ) + b.properties_byproduct[
+                t
+            ].get_material_flow_terms(
+                "Liq", "H2O"
+            )
+
+        # default water recovery
+        @self.Constraint(self.flowsheet().time, doc="Default water recovery equation")
+        def default_water_recovery_equation(b, t):
+            return b.recovery_frac_mass_H2O[t] == 1
+
+        # Solute removal
+        @self.Constraint(
+            self.flowsheet().time,
+            self.config.property_package.phase_list,
+            self.config.property_package.solute_set,
+            doc="Solute removal equations",
+        )
+        def solute_removal_equation(b, t, p, j):
+            return b.removal_frac_mass_comp[t, j] * b.properties_in[
+                t
+            ].get_material_flow_terms(p, j) == b.properties_byproduct[
+                t
+            ].get_material_flow_terms(
+                p, j
+            )
+
+        # Solute concentration of treated stream
+        @self.Constraint(
+            self.flowsheet().time,
+            self.config.property_package.phase_list,
+            self.config.property_package.solute_set,
+            doc="Constraint for solute concentration in treated " "stream.",
+        )
+        def solute_treated_equation(b, t, p, j):
+            return (1 - b.removal_frac_mass_comp[t, j]) * b.properties_in[
+                t
+            ].get_material_flow_terms(p, j) == b.properties_treated[
+                t
+            ].get_material_flow_terms(
+                p, j
+            )
+
+        # Default solute concentration
+        @self.Constraint(
+            self.flowsheet().time,
+            self.config.property_package.solute_set,
+            doc="Default solute removal equations",
+        )
+        def default_solute_removal_equation(b, t, j):
+            if hasattr(self.config.property_package.solute_set, "S_PO4"):
+                return b.removal_frac_mass_comp[t, "S_PO4"] == 0.98
+            elif hasattr(self.config.property_package.solute_set, "S_NH4"):
+                return b.removal_frac_mass_comp[t, "S_NH4"] == 0.3
```

Comment:
Is it worth considering S_N2 and S_NO3 here as well or are we just assuming that most nitrogen will be present in the form of "Ammonium plus ammonia nitrogen"?

Reply:
Alright, then I agree.